### PR TITLE
JEP-305 demo

### PR DESCRIPTION
**Pull request by jglick**
For now, just a demo of https://github.com/jenkinsci/plugin-pom/pull/100.

### CODE REVIEW

File: src/main/java/hudson/plugins/git/UserMergeOptions.java [hunk 1]
```diff
@@ -121,8 +121,8 @@ public String toString() {
         return "UserMergeOptions{" +
                 "mergeRemote='" + mergeRemote + '\'' +
                 ", mergeTarget='" + mergeTarget + '\'' +
-                ", mergeStrategy='" + mergeStrategy + '\'' +
-                ", fastForwardMode='" + fastForwardMode + '\'' +
+                ", mergeStrategy='" + getMergeStrategy().name() + '\'' +
+                ", fastForwardMode='" + getFastForwardMode().name() + '\'' +
```
Comment (jglick) [inline src/main/java/hudson/plugins/git/UserMergeOptions.java]: Follow-up to #580.

**Comment by jglick** [comment]
Would need a squash-merge but this is actually safe to merge now, I think. I will show how to consume its artifact downstream.

**Comment by jglick** [comment]
Should be unrelated.

**Comment by jglick** [comment]
> new enforcer rules that takes days to fix

Any _existing_ problems I have already dealt with here.

**Comment by rsandell** [comment]
Well my PR contains a bunch of plugin dependency bumps that I already spent a day tweaking until the enforcer got happy :)

**Comment by raul-arabaolaza** [comment]
@jglick It seems [JENKINS-50790](https://issues.jenkins-ci.org/browse/JENKINS-50790) is hitting again, I have reopened and will work on it. This is a flakiness (as there are builds where it passes) due to load I believe.

**Comment by MarkEWaite** [comment]
@jglick since the tests have passed, is this ready to merge into the master branch?

**Comment by jglick** [comment]
@MarkEWaite yes it could be merged any time.

Note that the `Jenkinsfile` is a bit messed up and actually deploys to Incrementals _before_ running PCT and ATH, which it should not be doing. I think @raul-arabaolaza and @oleg-nenashev know about this issue; requires some refactorings to `pipeline-library`. Strangely, Jenkins also seems to be setting the regular commit status before the build is actually finished, and I do not know why.

**Comment by jglick** [comment]
> will I need to use a different technique to create new releases?

See the [guide](https://github.com/jenkinsci/incrementals-tools/blob/master/README.md#running-maven-releases).

**Comment by jglick** [comment]
Hmm, getting timeouts in `PluginManager.installPlugins` again, not sure why. CC @raul-arabaolaza

**Comment by jglick** [comment]
Mirror timeout, retrying.

**Pull request merged**